File: src/main/java/com/cpw/browser/editor/BrowserEditorTabTitleProvider.java
package com.cpw.browser.editor;

import com.intellij.openapi.fileEditor.impl.EditorTabTitleProvider;
import com.intellij.openapi.project.Project;
import com.intellij.openapi.vfs.VirtualFile;

// 浏览器编辑器标签页标题提供者，截断过长标题并追加省略号
public class BrowserEditorTabTitleProvider implements EditorTabTitleProvider {

    // 获取编辑器标签页标题，超过 15 个字符时截断
    @Override
    public String getEditorTabTitle(Project project, VirtualFile file) {
        // 仅处理浏览器文件类型
        if (file.getFileType() == BrowserFileType.INSTANCE) {
            String title = file.getUserData(BrowserFileEditor.TITLE_KEY);
            // 无标题时返回默认值
            if (title == null) {
                return "Web Browser";
            }
            // 标题超过 15 个字符时截断并追加"..."
            return title.length() > 15 ? title.substring(0, 15) + "..." : title;
        }
        return null;
    }
}

File: src/main/kotlin/com/cpw/browser/editor/BrowserEditorTabTitleProvider.kt
package com.cpw.browser.editor

import com.intellij.openapi.fileEditor.impl.EditorTabTitleProvider
import com.intellij.openapi.project.Project
import com.intellij.openapi.vfs.VirtualFile

class BrowserEditorTabTitleProvider : EditorTabTitleProvider {

    override fun getEditorTabTitle(project: Project, file: VirtualFile): String? {
        if (file.fileType == BrowserFileType) {
            val title = file.getUserData(BrowserFileEditor.TITLE_KEY)
                ?: return "Web Browser"
            return if (title.length > 15) title.take(15) + "..." else title
        }
        return null
    }
}
